src/index.js

```diff
@@ -175,6 +171,39 @@ async function check(autoCheckElement: AutoCheckElement) {
   }
 }
 
+async function processResponse(response: Response, input: HTMLInputElement, required: boolean) {
+  if (response.status === 200) {
+    if (required) {
+      input.setCustomValidity('')
+    }
+    input.dispatchEvent(
+      new CustomEvent('auto-check-success', {
+        bubbles: true,
+        detail: {
+          response: response.clone()
+        }
+      })
+    )
+    return
+  }
+
+  const messages = []
+  const setValidity = result => messages.push(result)
+  input.dispatchEvent(
+    new CustomEvent('auto-check-error', {
+      bubbles: true,
+      detail: {
+        response: response.clone(),
+        setValidity
```

Comment:
What do you think about not passing `setValidity` if not `autocheck.required`? since we won't use it. Extracting the dispatching out to a function to be shared by `auto-check-send` and `auto-check-error` might make this easier.

Reply:
We want to limit `null` values so the payload has a reliable set of keys that listeners can use. If it's possibly null, then listeners must check before calling.

```js
input.addEventListener('auto-check-error', function(event) {
  if (event.detail.setValidity) {
    event.detail.setValidity('Hello')
  }
})

```